x-pack/filebeat/input/gcs/metrics.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package gcs
+
+import (
+	"github.com/rcrowley/go-metrics"
+
+	"github.com/elastic/beats/v7/libbeat/monitoring/inputmon"
+	"github.com/elastic/elastic-agent-libs/monitoring"
+	"github.com/elastic/elastic-agent-libs/monitoring/adapter"
+)
+
+// inputMetrics handles the input's metric reporting.
+type inputMetrics struct {
+	unregister        func()
+	url               *monitoring.String // URL of the input resource
+	errorsTotal       *monitoring.Uint   // number of errors encountered
+	decodeErrorsTotal *monitoring.Uint   // number of decode errors encountered
+
```

Comment:
Also, is possible to create a `source_lag_time` histogram metric based on the `updated` metadata from the GCS object? e.g.

- `source_lag_time` - Histogram of the difference between the [object modification time](https://cloud.google.com/storage/docs/metadata#modification-time) and the time when the object was read expressed in nanoseconds.

Reply:
Will implement this